python/cuml/dask/common/part_utils.py

```diff
@@ -128,24 +129,23 @@ def _extract_partitions(dask_obj, client=None):
 
     client = default_client() if client is None else client
 
-    # dask_cudf.dataframe or dask_cudf.series or dask.array
-    if (isinstance(dask_obj, dcDataFrame) or
-            isinstance(dask_obj, daskArray) or
-            isinstance(dask_obj, dcSeries)):
-        parts = futures_of(client.persist(dask_obj))
+    # dask.dataframe or dask.array
+    if isinstance(dask_obj, (dcDataFrame, daskArray, daskSeries)):
+        persisted = client.persist(dask_obj)
+        parts = futures_of(persisted)
 
     # iterable of dask collections (need to colocate them)
     elif isinstance(dask_obj, Sequence):
         # NOTE: We colocate (X, y) here by zipping delayed
         # n partitions of them as (X1, y1), (X2, y2)...
         # and asking client to compute a single future for
         # each tuple in the list
-        parts = [d.to_delayed().ravel() if isinstance(d, daskArray)
-                 else d.to_delayed() for d in dask_obj]
-        parts = client.compute([p for p in zip(*parts)])
+        dela = [np.asarray(d.to_delayed()) for d in dask_obj]
```

Comment:
do we know at this point that these should all be host-side arrays?

Reply:
Yeah, `d.to_delayed()` returns a list of delayed objects (1 per underlying Dask partition) on host.